src/RGBPatternManager.cpp

```diff
@@ -0,0 +1,136 @@
+#include "RGBPatternManager.h"
+
+#include "Logging.h"
+
+#include <Arduino.h>
+
+const char* const TAG = "RGBPatternManager";
+
+using namespace OpenShock;
+
+// Currently this assumes a single WS2812B LED
+// TODO: Support multiple LEDs ?
+// TODO: Support other LED types ?
+
+RGBPatternManager::RGBPatternManager(std::uint8_t rgbPin) : m_rgbPin(rgbPin), m_rgbBrightness(255), m_name {0}, m_pattern(nullptr), m_patternLength(0), m_taskHandle(nullptr), m_taskSemaphore(xSemaphoreCreateBinary()) {
+  snprintf(m_name, sizeof(m_name), "RGBPatternManager-%u", m_rgbPin);
+
+  if ((m_rmtHandle = rmtInit(m_rgbPin, RMT_TX_MODE, RMT_MEM_64)) == NULL) {
+    ESP_LOGE(m_name, "Failed to create rgb rmt object");
```

Comment:
Please only use TAG for ESP_LOGX calls, i made this mistake in the past but have corrected it 😅